common/src/main/java/com/microsoft/identity/common/internal/cache/CacheKeyValueDelegate.java

```diff
@@ -192,6 +196,9 @@ public String generateCacheKey(Credential credential) {
             final IdTokenRecord idToken = (IdTokenRecord) credential;
             cacheKey = cacheKey.replace(REALM, sanitizeNull(idToken.getRealm()));
             cacheKey = cacheKey.replace(TARGET, "");
+        } else if (credential instanceof PrimaryRefreshTokenRecord) {
+            cacheKey = cacheKey.replace(REALM, "");
```

Comment:
I don't understand this. Why is this value being set to `""`? Can you point me towards where this is outlined in the schema spec?

Reply:
OK -  I believe this comes from [here](https://identitydivision.visualstudio.com/DevEx/_git/AuthLibrariesApiReview?path=%2FSSO%2FSchema.md&version=GBiambmelt%2Fclient_server_nonce&line=241&lineEnd=242&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents) - can you please update the implementation of [`AbstractAccountCredentialCache#getCredentialsFilteredByInternal()`](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/dev/common/src/main/java/com/microsoft/identity/common/internal/cache/AbstractAccountCredentialCache.java#L125) to ensure this is handled correctly when filtering serialized records? It looks like PRT lookups need to behave like wildcards when filtering by `realm` and `target`. See uses of `mustMatchOnRealm` and `mustMatchOnTarget`

Reply:
Just looked `mustMatchOnTarget`, `mustMatchOnRealm`, it works as wildcard for PRT.

Reply:
Also updated `mustMatchOnClientId` to not filter by client_id for PRT.

Reply:
> Also updated `mustMatchOnClientId` to not filter by client_id for PRT, and added two tests.

@richard1122 I no longer see these changes in this PR. Not needed?

Reply:
Hi @iambmelt , from discussion [here](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1177#discussion_r564185002), we will continue support filter by client id, and let caller decide.

Reply:
ACK, resolved